src/main/java/org/terracotta/statistics/derived/latency/DefaultLatencyHistogramStatistic.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * All content copyright Terracotta, Inc., unless otherwise indicated.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.statistics.derived.latency;
+
+import org.terracotta.statistics.Time;
+import org.terracotta.statistics.derived.histogram.BarSplittingBiasedHistogram;
+import org.terracotta.statistics.derived.histogram.Histogram;
+import org.terracotta.statistics.observer.ChainedEventObserver;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.LongSupplier;
+
+import static java.lang.Math.nextDown;
+
+/**
+ * @author Mathieu Carbou
+ */
+public class DefaultLatencyHistogramStatistic implements LatencyHistogramStatistic, ChainedEventObserver {
+
+  private final BarSplittingBiasedHistogram histogram;
+  private final LongSupplier timeSupplier;
+  private final AtomicLong nextPruning = new AtomicLong();
+  private final long pruningDelay;
+
+  /**
+   * Create a histogram maintained over a sliding time window.
+   * <p>
+   * The constructed histogram is:
+   * </p>
+   * <ul>
+   * <li>maintained over {@code window} sliding window</li>
+   * <li>consists of {@code bucketCount} buckets</li>
+   * <li>where {@code b1.size() ~= b0.size * phi}</li>
+   * </ul>
+   * If "phi" is high, the quantile bounds will be more precise for lower percentiles such as minimum.
+   * If "phi" is low, the quantile bounds will be more precise for higher percentiles such as 99%-ile, maximum.
+   *
+   * @param phi         histogram bucket bias factor
+   * @param bucketCount number of buckets
+   * @param window      sliding window size, in ns
+   * @param timeSupplier      the supplier of time, which must be in the same unit as the time passed to the {{@link #event(long, long)}} method.
+   */
+  public DefaultLatencyHistogramStatistic(
+      double phi,
+      int bucketCount,
+      Duration window,
+      LongSupplier timeSupplier) {
+    this.timeSupplier = timeSupplier;
+    this.histogram = new BarSplittingBiasedHistogram(phi, bucketCount, window.toNanos());
+    this.pruningDelay = window.toNanos() / 2;
```

Comment:
@chrisdennis : I've set the pruning delay (when inserting only) to be half of the window to expire the histogram. Do you think it should be done more often ?